ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -334,19 +386,85 @@ where is_in_physics_bubble(vessel)
   }
 
   private void Update() {
-    if (fix_nav_ball_in_plotting_frame &&
-        has_active_vessel_in_space()) {
+    override_rsas_target_ = false;
+    Vessel active_vessel = FlightGlobals.ActiveVessel;
+    if (active_vessel != null &&
+        !FlightGlobals.ActiveVessel.isEVA) {
       if (nav_ball_ == null) {
         nav_ball_ = (NavBall)FindObjectOfType(typeof(NavBall));
       }
-      nav_ball_.navBall.rotation =
-          (UnityEngine.QuaternionD)nav_ball_.attitudeGymbal *  // sic.
-              (UnityEngine.QuaternionD)NavBallOrientation(
-                  plugin_,
-                  transforms_,
-                  (XYZ)Planetarium.fetch.Sun.position,
-                  (XYZ)(Vector3d)
-                      FlightGlobals.ActiveVessel.ReferenceTransform.position);
+      if (compass_nav_ball_texture_ == null) {
+        compass_nav_ball_texture_ =
+            nav_ball_.navBall.renderer.material.mainTexture;
+      }
+
+      if (nav_ball_changed_) {
+        // Texture the ball.
+        nav_ball_changed_ = false;
+        if (!fix_nav_ball_in_plotting_frame || !PluginRunning()) {
+          nav_ball_.navBall.renderer.material.mainTexture =
+              compass_nav_ball_texture_;
```

Comment:
Shouldn't you revert to the stock navball when the plugin is not running?  It seems odd to leave your asset loaded in that case.